### ISSUE TITLE
pachctl deploy fails with k8s 1.16.0

### ISSUE THREAD

**Issue by pappasilenus** [pachyderm/pachyderm#4154]
**What happened?**:
Updated to minikube 1.40. The standard deploy uses k8s 1.16.0.  A local deploy fails:
```
$ pachctl deploy local
serviceaccount/pachyderm configured
clusterrole.rbac.authorization.k8s.io/pachyderm configured
clusterrolebinding.rbac.authorization.k8s.io/pachyderm configured
service/etcd configured
service/pachd configured
service/dash configured
secret/pachyderm-storage-secret configured
unable to recognize "STDIN": no matches for kind "Deployment" in version "apps/v1beta1"
unable to recognize "STDIN": no matches for kind "Deployment" in version "apps/v1beta1"
unable to recognize "STDIN": no matches for kind "Deployment" in version "apps/v1beta1"
kubectl apply -f - --validate=false --namespace default: exit status 1
unable to recognize "STDIN": no matches for kind "Deployment" in version "apps/v1beta1"
unable to recognize "STDIN": no matches for kind "Deployment" in version "apps/v1beta1"
unable to recognize "STDIN": no matches for kind "Deployment" in version "apps/v1beta1"
```

**What you expected to happen?**:
I expected it to deploy

**How to reproduce it (as minimally and precisely as possible)?**:
1. update to minikube 1.40 
2. create a cluster using the default for `--kubernetes-version="1.16.0"`
3. `pachctl deploy local`

**Anything else we need to know?**:

A workaround may be to use an earlier version of k8s via that flag, I'm testing that now.

**Environment?**:
- Kubernetes version (use `kubectl version`): 1.16.0
- Pachyderm CLI and pachd server version (use `pachctl version`): 1.9.5
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s): minikube 1.40
- OS (e.g. from /etc/os-release): macOS 10.14.6
- Others:

**Comment by pappasilenus** [comment]
Confirmed using k8s v1.14.6 is a valid workaround.

**Comment by danielefranceschi** [comment]
Deployment in k8s 1.16 has fully transitioned to `apps/v1`, other apiversions were deprecated and now are **not supported** (see https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/).

Latest k8s version compatible with pachyderm 1.9.* is `1.15.*` (using it)

**Comment by pappasilenus** [comment]
It looks like a valid workaround would be to do the following:
```
$ pachctl deploy local --dry-run | jq 'select(.kind == "Deployment").apiVersion = "v1"' > /tmp/localmanifest.json
$ kubectl apply -f /tmp/localmanifest.json
```

And then, to work around the issue with pachyderm contexts until https://github.com/pachyderm/pachyderm/pull/4152 is merged and released, you'll have to do something like this to create the appropriate pachyderm config:

```
# if this doesn't work, add --overwrite to the end
$ echo `{"cluster_name": "minikube, "auth_info": "minikube", "namespace": "default"}` | pachctl config set context "minikube"
Reading from stdin

$ pachctl config set active-context minikube
```

**Comment by brokenjacobs** [comment]
can anyone confirm that pachyderm works on 1.16? The deployment is one thing but pachd deploys replicasets, presumably with the deprecated api?

**Comment by danielefranceschi** [comment]
@brokenjacobs pipeline workers are deployed via `v1/ReplicationController`, no problems IMO.

**Comment by danielefranceschi** [comment]
@pappasilenus `apps/v1` is available since v1.9, so maybe the question could be _dropping compatibility with k8s versions < 1.9_.

(my 2 cents: 1.16 support is a must)
